lms/static/scripts/frontend_apps/utils/jstor.js

```diff
@@ -1,52 +1,81 @@
 /**
- * Extract the JSTOR article ID from a URL.
+ * Test whether a value looks like a DOI.
  *
- * Accepts URLs of the form:
- * - http[s]://www.jstor.org/stable/<articleID> OR
- * - http[s]://www.jstor.org/stable/<doiPrefix>/<doiSuffix>
+ * See https://en.wikipedia.org/wiki/Digital_object_identifier#Nomenclature_and_syntax.
  *
- * (Query string is ignored on provided URLs)
+ * @param {string} value
+ */
+function isDOI(value) {
+  return /^10.([0-9]+\.?)+\/.*/.test(value);
+}
+
+/**
+ * Extract the JSTOR article ID or DOI from a form field value.
+ *
+ * Accepts input in various forms:
  *
- * and returns, respectively:
- * - <articleID> OR
- * - <doiPrefix>/<doiSuffix>
+ * - A numeric ID (1234)
+ * - A DOI (10.2307/1234)
+ * - A canonical URL for a JSTOR article (eg. http[s]://www.jstor.org/stable/<articleID>
+ *   OR http[s]://www.jstor.org/stable/<doiPrefix>/<doiSuffix>)
+ * - A URL for a JSTOR article that has being proxied through an institution's
+ *   EZProxy or similar service (eg. https://www-jstor-org.myuni.edu/stable/1234).
  *
- * Return `null` if provided string is not a URL or does not match one of the
- * accepted formats.
+ * Return `null` if the input does not match any of the recognized formats.
```

Comment:
If I'm not mistaken (i.e. overlooking it somewhere here), I don't think the JSTOR article ID format is ever explicitly described in the comments in this module. It's expressed in a regular expression on line 31, but not in prose. The function comments here include a "numeric ID" format, but the "Plain JSTOR article ID" regexp and module tests for JSTOR IDs include alpha characters and dots/periods. Can you clarify in comments?

Reply:
The comment about a "numeric ID" was wrong. I initially thought that all JSTOR article IDs which are not DOIs, are numeric. However that is not the case.